src/System.Management.Automation/engine/SessionStateLocationAPIs.cs

```diff
@@ -231,10 +232,36 @@ internal PathInfo SetLocation(string path, CmdletProviderContext context)
             ProviderInfo provider = null;
             string providerId = null;
 
+            // Replace path with last working directory when '-' was passed.
+            if (_workingLocationHistoryStack == null)
+            {
+                const int locationHistoryLimit = 1000;
+                _workingLocationHistoryStack = new BoundedStack<PathInfo>(locationHistoryLimit);
```

Comment:
Why here? It's always needed, so you can create in the constructor or where the field is declared.

Reply:
In terms of class design I would definitely agree but given that this partial class is spread across a lot of files, I thought it might be better to rather place it nearby its usage. I can of course move it into the constructor if that's your preference.

Reply:
If you initialize on the field, it's in the same file and relatively close to the usage.

As for preference - when I see the `if` test, I ask myself - how can this be `null`? I think most people don't do this though, they just assume it's a possible condition and move on, but I dig deeper. It doesn't feels like initialization because it's not obvious - e.g. deferred initialization in a property getter. So I investigate - and I shouldn't need to do that.

Also - the `if` test felt unnecessary (and hence, not good for performance, not that it matters here) - another reason to change it.

Reply:
Ok. I'll do it on the property then, which sounds like reasonable compromise.

Reply:
Uh, sorry, that's not what I was looking for.

Deferred initialization is done when something is expensive and not always needed. If something is always needed (should never be null), then it should be initialized in the constructor.